bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/BigtableSyncTableAdapter.java

```diff
@@ -0,0 +1,384 @@
+      jobConf.set(HConstants.ZOOKEEPER_ZNODE_PARENT, zkClusterKey.getZnodeParent());
+    }
+
+    // enable dry run as default
+    jobConf.setBoolean(
+        SyncTable.DRY_RUN_CONF_KEY, jobConf.getBoolean(SyncTable.DRY_RUN_CONF_KEY, true));
+
+    // Set up bigtable configurations for job. Note that the job conf is shared for source and
+    // target databases and additional configuration in mapper initializes proper
+    // connections.
+    if (sourceBigtableProjectId != null && sourceBigtableInstance != null) {
+      if (syncTable.sourceZkCluster != null)
+        LOG.warn(
+            "sourceZkCluster config("
+                + syncTable.sourceZkCluster
+                + ") overriden with sourceBigtableProjectId("
+                + sourceBigtableProjectId
+                + "), sourceBigtableInstance("
+                + sourceBigtableInstance
+                + ")");
+
+      jobConf.set(SOURCE_BT_PROJECTID_CONF_KEY, sourceBigtableProjectId);
+      jobConf.set(SOURCE_BT_INSTANCE_CONF_KEY, sourceBigtableInstance);
+
+      if (sourceBigtableAppProfile != null)
+        jobConf.set(SOURCE_BT_APP_PROFILE_CONF_KEY, sourceBigtableAppProfile);
+    }
+    if (targetBigtableProjectId != null && targetBigtableInstance != null) {
+      if (syncTable.targetZkCluster != null)
+        LOG.warn(
+            "targetZkCluster config("
+                + syncTable.targetZkCluster
+                + ") overridden with targetBigtableProjectId("
+                + targetBigtableProjectId
+                + "), targetBigtableInstance("
+                + targetBigtableInstance
+                + ")");
+
+      jobConf.set(TARGET_BT_PROJECTID_CONF_KEY, targetBigtableProjectId);
+      jobConf.set(TARGET_BT_INSTANCE_CONF_KEY, targetBigtableInstance);
+
+      if (targetBigtableAppProfile != null)
+        jobConf.set(TARGET_BT_APP_PROFILE_CONF_KEY, targetBigtableAppProfile);
+
+      BigtableConfiguration.configure(
+          jobConf,
+          jobConf.get(TARGET_BT_PROJECTID_CONF_KEY),
+          jobConf.get(TARGET_BT_INSTANCE_CONF_KEY),
+          jobConf.get(TARGET_BT_APP_PROFILE_CONF_KEY, ""));
+    }
+
+    Scan jobScan = TableMapReduceUtil.convertStringToScan(jobConf.get(TableInputFormat.SCAN));
+    TableMapReduceUtil.initTableMapperJob(
+        syncTable.targetTableName, jobScan, BigtableSyncMapper.class, null, null, job);
+
+    return job;
+  }
+
+  /**
+   * parse and set command line options and args
+   *
+   * @param syncTable
+   * @param NUM_ARGS
+   * @param args
+   * @return
+   */
+  public static boolean parseCommandLine(SyncTable syncTable, int NUM_ARGS, final String[] args) {
+    // set any defaults & override with any options/args that are set on cli
+    setDefaultConfigs(syncTable);
+
+    if (args.length < NUM_ARGS) {
+      printUsage(null, args);
+      return false;
+    }
+    try {
+      syncTable.sourceHashDir = new Path(args[args.length - 3]);
+      syncTable.sourceTableName = args[args.length - 2];
```

Comment:
Where is the args coming from? Where do we enforce strict ordering of the arguments?

Reply:
we discussed the input args, i don't believe any changes are required here, but let me know.